lib/tasks/puppet.rake

```diff
@@ -46,7 +46,7 @@ namespace :puppet do
 
         Host.find_each do |host|
           $stdout.print "processing #{host.name} "
-          nodeinfo = YAML.load `#{script} #{host.name}`
+          nodeinfo = YAML.safe_load `#{script} #{host.name}`
           if nodeinfo.is_a?(Hash)
             $stdout.puts "DONE" if host.importNode nodeinfo
```

Comment:
Shall we drop the task instead? `importNode` was removed in 9584548fbc6188fefd462b6593384aed0d8c20f1. If we want to keep this task (which I seriously doubt), then it should live in foreman_puppet instead.

Reply:
For easier cherry-picking, I think we should do this in a separate PR, opened Redmine for it: https://projects.theforeman.org/issues/36222
cc @nadjaheitmann

Reply:
You are correct that it deserves its own commit and own issue, but if I'm honest, I feel safer to cherry pick the removal than using `YAML.safe_load`. We know this task is broken since Foreman 3.0 so it's broken anyway. And if you have the permission to call a rake task, you're already in full control of Foreman that you could also call `foreman-rake console`. Using `YAML.safe_load` can break things and https://github.com/rails/rails/commit/611990f1a6c137c2d56b1ba06b27e5d2434dcd6a certainly did.

Note I'm not saying we shouldn't pick, because security can be more important but we should be careful about it.

Reply:
OK then, dropping it in this PR as a separate commit,
I don't mind cherry-picking it as well